frontend/web/src/assets/lib/jsc3d/jsc3d.webgl.js

```diff
@@ -0,0 +1,1202 @@
+			}
+			edges = new Float32Array(edges);
+		}
+
+		return edges;
+	}
+
+	var gl = this.gl;
+
+	var needFlat = (renderMode == 'flat') || (renderMode == 'textureflat');
+
+	var hasTrianglesOnly = mesh.indexBuffer.length == 4 * mesh.faceCount;
+	var hasTextureCoords = mesh.texCoordBuffer && mesh.texCoordBuffer.length >= 2;
+
+	var ibuf  = mesh.indexBuffer;
+	var vbuf  = mesh.vertexBuffer;
+	var tbuf  = mesh.texCoordBuffer;
+	var tibuf = mesh.texCoordIndexBuffer || ibuf;
+	var nbuf  = mesh.vertexNormalBuffer;
+	var nibuf = mesh.vertexNormalIndexBuffer || ibuf;
+	var fnbuf = mesh.faceNormalBuffer;
+	var numOfFaces = mesh.faceCount;
+
+	if (!mesh.compiled) {
+		/*
+		 * Rebuild all primitives from scratch.
+		 */
+
+		mesh.compiled = {
+			isIndexed: false
+		};
+
+		mesh.compiled.pickingId = new Float32Array([
+			(mesh.internalId & 0xff0000) / 16777216, (mesh.internalId & 0xff00) / 65536, (mesh.internalId & 0xff) / 256
+		]);
+
+		var triangles, edges, coords, normals, texcoords;
+		if(hasTrianglesOnly) {
+			coords = new Float32Array(9 * numOfFaces);
+			normals = new Float32Array(9 * numOfFaces);
+			if(hasTextureCoords)
+				texcoords = new Float32Array(6 * numOfFaces);
+			var v0, v1, v2;
+			var n0, n1, n2;
+			var t0, t1, t2;
+			for(var i=0, j=0, k=0, faceIndex=0; i<ibuf.length; i+=4, j+=9, k+=6, faceIndex++) {
+				v0 = ibuf[i    ] * 3;
+				v1 = ibuf[i + 1] * 3;
+				v2 = ibuf[i + 2] * 3;
+				coords[j    ] = vbuf[v0    ];
+				coords[j + 1] = vbuf[v0 + 1];
+				coords[j + 2] = vbuf[v0 + 2];
+				coords[j + 3] = vbuf[v1    ];
+				coords[j + 4] = vbuf[v1 + 1];
+				coords[j + 5] = vbuf[v1 + 2];
+				coords[j + 6] = vbuf[v2    ];
+				coords[j + 7] = vbuf[v2 + 1];
+				coords[j + 8] = vbuf[v2 + 2];
+
+				if(needFlat) {
+					n0 = faceIndex * 3;
+					normals[j    ] = normals[j + 3] = normals[j + 6] = fnbuf[n0    ];
+					normals[j + 1] = normals[j + 4] = normals[j + 7] = fnbuf[n0 + 1];
+					normals[j + 2] = normals[j + 5] = normals[j + 8] = fnbuf[n0 + 2];
+					
+				}
+				else {
+					n0 = nibuf[i    ] * 3;
+					n1 = nibuf[i + 1] * 3;
+					n2 = nibuf[i + 2] * 3;
+					normals[j    ] = nbuf[n0    ];
+					normals[j + 1] = nbuf[n0 + 1];
+					normals[j + 2] = nbuf[n0 + 2];
+					normals[j + 3] = nbuf[n1    ];
+					normals[j + 4] = nbuf[n1 + 1];
+					normals[j + 5] = nbuf[n1 + 2];
+					normals[j + 6] = nbuf[n2    ];
+					normals[j + 7] = nbuf[n2 + 1];
+					normals[j + 8] = nbuf[n2 + 2];
+				}
+
+				if(hasTextureCoords) {
+					t0 = tibuf[i    ] * 2;
+					t1 = tibuf[i + 1] * 2;
+					t2 = tibuf[i + 2] * 2;
+					texcoords[k    ] = tbuf[t0    ];
+					texcoords[k + 1] = tbuf[t0 + 1];
+					texcoords[k + 2] = tbuf[t1    ];
+					texcoords[k + 3] = tbuf[t1 + 1];
+					texcoords[k + 4] = tbuf[t2    ];
+					texcoords[k + 5] = tbuf[t2 + 1];
+				}
+			}
+		}
+		else {
+			coords = [];
+			normals = [];
+			if(hasTextureCoords)
+				texcoords = [];
+			var v0, v1, v2;
+			var n0, n1, n2;
+			var t0, t1, t2;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "t1" as this name is already used in declaration at line 926. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "t2" as this name is already used in declaration at line 926. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "t0" as this name is already used in declaration at line 926. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)